packages/quantic/force-app/main/default/lwc/quanticSearchBoxInput/__tests__/quanticSearchBoxInput.test.js

```diff
@@ -0,0 +1,344 @@
+    element[key] = value;
+  }
+  document.body.appendChild(element);
+  return element;
+}
+
+// Helper function to wait until the microtask queue is empty.
+function flushPromises() {
+  return new Promise((resolve) => setTimeout(resolve, 0));
+}
+
+describe('c-quantic-search-box-input', () => {
+  function cleanup() {
+    // The jsdom instance is shared across test cases in a single file so reset the DOM
+    while (document.body.firstChild) {
+      document.body.removeChild(document.body.firstChild);
+    }
+  }
+
+  afterEach(() => {
+    cleanup();
+    jest.clearAllMocks();
+  });
+
+  describe('when the textarea property is set to false', () => {
+    it('should display the search input properly', async () => {
+      const element = createTestComponent();
+      await flushPromises();
+
+      const input = element.shadowRoot.querySelector(selectors.searchBoxInput);
+      const submitButton = element.shadowRoot.querySelector(
+        selectors.searchBoxSubmitBtn
+      );
+      const clearIcon = element.shadowRoot.querySelector(
+        selectors.searchBoxClearIcon
+      );
+
+      expect(input).not.toBeNull();
+      expect(input.placeholder).toEqual(defaultOptions.placeholder);
+      expect(submitButton).not.toBeNull();
+      expect(clearIcon).toBeNull();
+    });
+  });
+
+  describe('when the textarea property is set to true', () => {
+    it('should display the search textarea properly', async () => {
+      const element = createTestComponent({...defaultOptions, textarea: true});
+      await flushPromises();
+
+      const submitButton = element.shadowRoot.querySelector(
+        selectors.searchBoxSubmitBtn
+      );
+      const clearIcon = element.shadowRoot.querySelector(
+        selectors.searchBoxClearIcon
+      );
+      const textarea = element.shadowRoot.querySelector(
+        selectors.searchBoxTextArea
+      );
+
+      expect(textarea).not.toBeNull();
+      expect(textarea.placeholder).toEqual(defaultOptions.placeholder);
+      expect(submitButton).not.toBeNull();
+      expect(clearIcon).toBeNull();
+    });
+  });
+
+  describe('when the withoutSubmitButton is set to false', () => {
+    it('should not display the searchIcon to the left of the searchbox', async () => {
+      const element = createTestComponent({
+        ...defaultOptions,
+        withoutSubmitButton: false,
+      });
+      await flushPromises();
+
+      const searchIcon = element.shadowRoot.querySelector(
+        selectors.searchBoxSearchIcon
+      );
+
+      expect(searchIcon).toBeNull();
+    });
+
+    it('should display the submit button to the right of the searchbox', async () => {
+      const element = createTestComponent({
+        ...defaultOptions,
+        withoutSubmitButton: false,
+      });
+      await flushPromises();
+
+      const submitButton = element.shadowRoot.querySelector(
+        selectors.searchBoxSubmitBtn
+      );
+
+      expect(submitButton).not.toBeNull();
+    });
+  });
```

Comment:
```suggestion
    it('should display the submit button correctly to the right only', async () => {
      const element = createTestComponent({
        ...defaultOptions,
        withoutSubmitButton: false,
      });
      await flushPromises();

      const submitButton = element.shadowRoot.querySelector(
        selectors.searchBoxSubmitBtn
      );
      const searchIcon = element.shadowRoot.querySelector(
        selectors.searchBoxSearchIcon
      );

      expect(searchIcon).toBeNull();
      expect(submitButton).not.toBeNull();
    });
  });
```

I would merge these two tests.